jerry-core/ecma/operations/ecma-async-generator-object.c

```diff
@@ -132,8 +293,6 @@ ecma_async_generator_finalize (vm_executable_object_t *async_generator_object_p,
   ecma_value_t next = async_generator_object_p->extended_object.u.class_prop.u.head;
   ecma_async_generator_task_t *task_p = ECMA_GET_INTERNAL_VALUE_POINTER (ecma_async_generator_task_t, next);
 
-  ECMA_SET_INTERNAL_VALUE_ANY_POINTER (async_generator_object_p->extended_object.u.class_prop.u.head, NULL);
```

Comment:
Note: setting head to NULL caused memory errors, since resolving promises may trigger gc, and promises referenced by tasks were not marked.